proxy/coco/src/request/waiting_room/stream.rs

```diff
@@ -85,9 +90,13 @@ impl Clones {
     /// exists then the stream does nothing and waits until the next poll. We note that the stream
     /// is infinite with this respect and will never return `None`.
     #[must_use]
-    pub fn new(waiting_room: Arc<RwLock<WaitingRoom<Instant, Duration>>>) -> Self {
+    pub fn new(
+        waiting_room: Arc<RwLock<WaitingRoom<Instant, Duration>>>,
+        duration: Duration,
```

Comment:
As mentioned above, it seems odd that this is configurable. It is internal to the proper implementation of the stream as it ensures that the waker is called and we make progress on this future.

Reply:
It's configurable because otherwise our tests would be stuck with the "reasonable" value which may be too long for them making our test suite slow.

Reply:
I'd argue the reasonable value is off. We should be able to poll at 100Hz without a sweat.

Reply:
I don't think I'm understanding. I picked a rate at random tbh. If you have a better default then I'm all ears :ear: However, I'm not sure why making it configurable is not ok

Reply:
Because it's the poll rate of the `impl Stream`, just an internal number to make sure the stream advances -- exclusively in the sense of feeding the async runtime. So it does not mean anything to the domain of the waitng_room/requests. It's not altering any domain behaviour. More so it provides a footgun, when missconfigured to potentially delay the progress of requests unnecessarily.

Reply:
But does not define how long the stream will wait before it tries to look for the next request? Like say the delay was set to 5 seconds, and our test missed the boat on checking the waiting room, then it has to wait for the next window, no? Adding a whole 5 seconds to our test suite :(

Reply:
There seems to be a miss-conception here. The delay should be in the order of max 10 millis. It's really a cheap solution to avoid a busy loop hogging the cpu while still making progress. We need this because we don't have a more effective way of signaling the waker passed in via context to `poll` that there is a `Ready` value. If we find a more effective, as in evented way, to signal the async runtime that we have a new value, this delay will vanish alltogether. It's a very deep implementation detail and should not be surfaced.

Reply:
Ya, I was completely misunderstanding :+1: Thanks for the patience :innocent: